src/store/reducers/UserProfile.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  UserProfileActions,
+  UserProfileActionTypes,
+} from "store/actions/UserProfile";
+import { User } from "types/User";
+import { WithId } from "utils/id";
+
+export type UserProfileState = {
+  userProfile: WithId<User> | undefined;
+};
+
+const initialChatState: UserProfileState = {
+  userProfile: undefined,
```

Comment:
You can just have an empty object here

Reply:
@mike-lvov Why do you suggest that rather than `undefined`? I'm not a fan of 'shadowing' 'empty' types with things like this (empty object, empty string, etc) where we can more correctly use `undefined`/etc to represent them (and then gain access to all of the related typing/etc tools that allow us to easily assert when things are 'empty in a direct way.

Reply:
You already have the type of this state a few lines above. I don't think of this as shadowing since you have direct access to the type of this state. But nvm, I don't feel strongly about this

Reply:
Yeah, but the type above is `WithId<User>` or `undefined`, and you're suggesting we default it to `{}` which is neither of those things. That's what i'm saying is the issue here.